utils/http_test.go

```diff
@@ -1,30 +1,25 @@
 package utils
 
 import (
-	"fmt"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"strings"
 	"testing"
 
-	"golang.org/x/net/context"
-
+	"github.com/docker/distribution/registry/api/errcode"
 	"github.com/endophage/gotuf/signed"
+	"golang.org/x/net/context"
 
 	"github.com/docker/notary/errors"
 )
 
-func MockContextHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) *errors.HTTPError {
+func MockContextHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) errcode.Errors {
 	return nil
 }
 
-func MockBetterErrorHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) *errors.HTTPError {
-	return &errors.HTTPError{
-		HTTPStatus: http.StatusInternalServerError,
-		Code:       9999,
-		Err:        fmt.Errorf("TestError"),
-	}
+func MockBetterErrorHandler(ctx context.Context, w http.ResponseWriter, r *http.Request) errcode.Errors {
+	return errcode.Errors{errors.ErrUnknown.WithDetail("Test Error")}
```

Comment:
@endophage Try this:

```
return errcode.Errors{
    errors.ErrUnknown.WithDetail("Test Error"),
}
```

@dsymonds Any ideas? This is a slice type that is failing a golint check for unkeyed fields on a struct literal:

```
utils/http_test.go:22: errcode.Errors composite literal uses unkeyed fields
```